src/manager.rs

```diff
@@ -22,9 +22,10 @@ struct WatcherWrapper {
 }
 
 // watcher is actually a Box<WatcherWrapper>
-extern "C" fn watcher_cb(notification: *const ExternNotification, watcher: *mut c_void) {
-    let watcher_wrapper: &mut WatcherWrapper = unsafe { &mut *(watcher as *mut WatcherWrapper) };
-    watcher_wrapper.watcher.on_notification(Notification::new(notification));
+extern "C" fn watcher_cb(notification: *const ExternNotification, watcher: *const c_void) {
+    let watcher_wrapper: &WatcherWrapper = unsafe { &*(watcher as *const WatcherWrapper) };
```

Comment:
fortunately it was a no-brainer to move to non-mut references because on_notification was taking &self and not &mut self :)